api/src/middleware/__tests__/competenciesRouter.ts

```diff
@@ -104,4 +109,80 @@ describe('competenciesRouter', () => {
         .expect(500, done);
     });
   });
+
+  describe('PUT /:id', () => {
+    it('should update a competency of user who authored it', done => {
+      const principalId = 2;
+      const competencyId = 3;
+      const label = 'label';
+      const description = 'description';
+      const competency = {
+        id: competencyId,
+      };
+      mockPrincipalId(principalId);
+      mockAuthorizeCompetencyUpdate.mockResolvedValue(true);
+      mockUpdateCompetency.mockResolvedValue();
+      appAgent
+        .put('/3')
+        .send({
+          label: label,
+          description: description,
+        })
+        .expect(200, itemEnvelope(competency), err => {
+          expect(mockAuthorizeCompetencyUpdate).toHaveBeenCalledWith(
+            principalId,
+            competencyId
+          );
+          expect(mockUpdateCompetency).toHaveBeenCalledWith(
+            competencyId,
+            label,
+            description
+          );
+          done(err);
+        });
+    });
+
+    it('should respond with a validation error if given id is less than 1', done => {
+      mockAuthorizeCompetencyUpdate.mockResolvedValue(true);
+      appAgent.put('/0').send({ label: '', description: '' }).expect(400, done);
+    });
+
+    it('should respond with an internal server error if an error was thrown while authorising a competency update', done => {
```

Comment:
Convention has the `internal server error` tests at the bottom of the `describe` blocks.  I think this test could be moved to the bottom as well.

Reply:
moved the test as suggested to the bottom.